projects/dex-ui/src/wells/useLiquidityQuote.tsx

```diff
@@ -1,65 +1,125 @@
 import { Well } from "@beanstalk/sdk/Wells";
 import { useQuery } from "@tanstack/react-query";
 import { Token, TokenValue } from "@beanstalk/sdk";
-import { useCallback } from "react";
+import { useMemo } from "react";
 import { LiquidityAmounts, REMOVE_LIQUIDITY_MODE } from "src/components/Liquidity/types";
+import { Log } from "src/utils/logger";
+import { useAccount } from "wagmi";
 
 export const useLiquidityQuote = (
   well: Well,
   removeLiquidityMode: REMOVE_LIQUIDITY_MODE,
   lpTokenAmount: TokenValue,
   singleTokenIndex: number,
   wellTokens: Token[],
-  amounts: LiquidityAmounts
+  amounts: LiquidityAmounts,
+  onQuoteHandler: () => void,
 ) => {
-  const atLeastOneAmountNonzero = useCallback(() => Object.values(amounts).filter((amount) => amount.value.gt("0")).length > 0, [amounts]);
+  const { address } = useAccount();
+
+  const bothAmountsNonZero = useMemo(
+    () => {
+      if (!well.tokens) {
+        return false;
+      }
+
+      if (well.tokens.length === 0) {
+        return false;
+      }
+
+      const nonZeroValues = Object.values(amounts).filter((amount) => amount.value.gt("0")).length;
+      
+      return nonZeroValues === well.tokens?.length;
+    },
+    [amounts, well.tokens]
+  );
+
+  Log.module("useLiquidityQuote").debug("amounts:", amounts);
+  Log.module("useLiquidityQuote").debug("bothAmountsNonZero:", bothAmountsNonZero);
+  Log.module("useLiquidityQuote").debug("removeLiquidityMode:", removeLiquidityMode);
```

Comment:
Can we combine this into one statement, with a more helpful message and an object containing all these props? ex:
```suggestion
  Log.module("useLiquidityQuote").debug("Quote details:", { amounts, bothAmountsNonZero, removeLiquidityMode });
```